### PR TITLE
add Universal Provider option

### PR DESCRIPTION
**Pull request by glitch-txs**
# Changes

This is a flag for internal use on our WalletAdapter client
The purpose is to filter out wallets that do not support WC protocol when using the universal provider

- feat: 
- [x] Add a "low" level flag for universal provider -> `isUniversalProvider`.
- [x]  Remove Browser option from wallets when `isUniversalProvider` is set to `true`.
- [x] Filter out injected-only wallets from "All Wallets" when `isUniversalProvider` is set to `true`.

closes https://github.com/WalletConnect/walletconnect-solana/issues/25

### CODE REVIEW

File: packages/core/src/controllers/ApiController.ts [hunk 1]
```diff
@@ -65,6 +65,14 @@ export const ApiController = {
     }
   },
 
+  _filterOutExtensions(wallets: WcWallet[]) {
+    if (OptionsController.state.isUniversalProvider) {
+      return wallets.filter(w => Boolean(w.mobile_link))
```
Comment (enesozturk) [inline packages/core/src/controllers/ApiController.ts:70]: Why we only enable mobile wallets if universal provider enabled?

Comment (glitch-txs) [reply packages/core/src/controllers/ApiController.ts]: updated to Mobile + Desktop + Webapps

**Review by enesozturk — APPROVED** [review]

File: packages/scaffold-ui/src/views/w3m-connecting-wc-view/index.ts [hunk 2]
```diff
@@ -133,7 +133,7 @@ export class W3mConnectingWcView extends LitElement {
     const { mobile_link, desktop_link, webapp_link, injected, rdns } = this.wallet
     const injectedIds = injected?.map(({ injected_id }) => injected_id).filter(Boolean) as string[]
     const browserIds = rdns ? [rdns] : injectedIds ?? []
-    const isBrowser = browserIds.length
+    const isBrowser = OptionsController.state.isUniversalProvider ? false : browserIds.length
```
Comment (tomiir) [inline packages/scaffold-ui/src/views/w3m-connecting-wc-view/index.ts:136]:
```suggestion
    const isBrowser = !OptionsController.state.isUniversalProvider || browserIds.length
```

File: packages/scaffold-ui/src/views/w3m-connecting-wc-view/index.ts [hunk 3]
```diff
@@ -133,7 +133,7 @@ export class W3mConnectingWcView extends LitElement {
     const { mobile_link, desktop_link, webapp_link, injected, rdns } = this.wallet
     const injectedIds = injected?.map(({ injected_id }) => injected_id).filter(Boolean) as string[]
     const browserIds = rdns ? [rdns] : injectedIds ?? []
-    const isBrowser = browserIds.length
+    const isBrowser = !OptionsController.state.isUniversalProvider || browserIds.length
```
Comment (glitch-txs) [inline packages/scaffold-ui/src/views/w3m-connecting-wc-view/index.ts:136]:
```suggestion
    const isBrowser = OptionsController.state.isUniversalProvider ? false : browserIds.length
```

**Pull request merged**